k8sdeps/kv/plugin/goplugin.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"plugin"
+)
+
+var _ Factory = &goFactory{}
+
+const (
+	dir = "$HOME/.config/kustomize/plugins/kvsource"
+)
+
+func init() {
```

Comment:
this is going to have to be removed, because we want to continue the notion of dependency injection, and no invisible side effects from inits, since this code is going to be used in kubectl.  There's no usage of inits in the current code.

 But for me to tell you how to inject one of these at the right place via github comments might take more time than just refactoring it later.  At the end of the day, the plugin registry will have to be a field in KustTarget, and get passed down in a call to generateConfigMapsAndSecrets.